src/main/java/lavalink/client/io/LavalinkRestClient.java

```diff
@@ -0,0 +1,230 @@
+        if (builderConsumer == null) return httpClientBuilder.build();
+
+        builderConsumer.accept(httpClientBuilder);
+        return httpClientBuilder.build();
+    }
+
+    private JSONObject apiGet(final String url, final String auth) throws IOException {
+        final HttpGet request = new HttpGet(url);
+        request.addHeader(HttpHeaders.AUTHORIZATION, auth);
+
+        final HttpClient httpClient = buildClient();
+        final HttpResponse httpResponse = httpClient.execute(request);
+        final int statusCode = httpResponse.getStatusLine().getStatusCode();
+        if (statusCode != 200) throw new IOException("Invalid API Request Status Code: " + statusCode);
+
+        final HttpEntity entity = httpResponse.getEntity();
+        if (entity == null) throw new IOException("Invalid API Response: No Content");
+
+        final String response = EntityUtils.toString(entity);
+        return new JSONObject(response);
+    }
+
+    private static final class TrackLoadResultHandler {
+
+        private final JSONObject loadResult;
+
+        private TrackLoadResultHandler(JSONObject loadResult) {
+            this.loadResult = loadResult;
+        }
+
+        private AudioTrack handleTrackLoaded() throws IOException {
+            final JSONArray trackDataSingle = loadResult.getJSONArray("tracks");
+            final JSONObject trackObject = trackDataSingle.getJSONObject(0);
+            final String singleTrackBase64 = trackObject.getString("track");
+
+            return LavalinkUtil.toAudioTrack(singleTrackBase64);
+        }
+
+        private AudioPlaylist handlePlaylistLoaded(boolean isSearchResult) throws Exception {
+            final JSONArray trackData = loadResult.getJSONArray("tracks");
+            final List<AudioTrack> tracks = new ArrayList<>();
+
+            for (final Object track : trackData) {
+                final String trackBase64 = ((JSONObject) track).getString("track");
+                final AudioTrack audioTrack = LavalinkUtil.toAudioTrack(trackBase64);
+
+                tracks.add(audioTrack);
+            }
+
+            if (isSearchResult) {
+                if (tracks.size() == 0) {
+                    throw new FriendlyException(
+                            "No search results found",
+                            FriendlyException.Severity.COMMON,
+                            new IllegalStateException("No results")
+                    );
+                }
+
+                return new BasicAudioPlaylist("Search results for: ", tracks, tracks.get(0), true);
+            }
+
+            final JSONObject playlistInfo = loadResult.getJSONObject("playlistInfo");
+            final int selectedTrackID = playlistInfo.getInt("selectedTrack");
+            final AudioTrack selectedTrack;
+
+            if (selectedTrackID < tracks.size() && selectedTrackID >= 0) {
+                selectedTrack = tracks.get(selectedTrackID);
+            } else {
+                if (tracks.size() == 0) {
```

Comment:
Is this actually something we want to check for?

Reply:
Actually, no. Another thing I kinda just blindly copied from @ByteAlex's code